packages/signature-controller/src/SignatureController.ts

```diff
@@ -902,6 +914,13 @@ export class SignatureController extends BaseController<
     request: OriginalRequest,
     chainId: string,
   ) {
+    if (
+      !this.#getFeatureFlags ||
+      this.#getFeatureFlags().disableDecodingApi ||
+      !this.#decodingApiUrl
```

Comment:
If we have an explicit toggle for decoding, can we make the URL required so we aren't complicating the logic with two conditions?

Reply:
That will be hard for instance on mobile side decoding is not yet implemented, thus client may not want to pass the URL

Reply:
But if that's the case, why even have a specific callback when the client could just dynamically specify no URL?

It's ultimately two options that achieve the same result.

Reply:
We need callback to read these preferences `useExternalServices and useTransactionSimulations`, as these can change on runtime.

Reply:
True, but I mean we could theoretically have a single `getDecodingApiUrl` callback instead, but perhaps the lesser of two evils to have a separate callback for clarity 👍